src/v2/expressions.ts

```diff
@@ -0,0 +1,82 @@
+import { Param } from './params/types';
+
+/*
+ * A CEL expression which can be evaulated during function deployment, and
+ * resolved to a value of the generic type parameter: i.e, you can pass
+ * an Expression<number> as the value of an option that normally accepts numbers.
+ */
+export abstract class Expression<
+  T extends string | number | boolean | string[]
+> {
+  rawValue: string;
+
+  get val(): any {
+    return this.rawValue;
+  }
+
+  toCEL() {
+    return `{{ ${this.toString()} }}`;
+  }
+}
+
+export class IfElseExpression<
+  T extends string | number | boolean | string[]
+> extends Expression<T> {
+  test: Expression<boolean>;
+  ifTrue: T;
+  ifFalse: T;
+
+  constructor(test: Expression<boolean>, ifTrue: T, ifFalse: T) {
+    super();
+    this.test = test;
+    this.ifTrue = ifTrue;
+    this.ifFalse = ifFalse;
+  }
+
+  toString() {
+    return `${this.test} ? ${this.ifTrue} : ${this.ifFalse}`;
+  }
+}
+
+export class CompareExpression<
```

Comment:
Nit: requesting docs on the functions in this class

Reply:
Done.